src/editor/edmap.cpp

```diff
@@ -86,6 +86,28 @@ std::optional<tile_index> CTileIconsSet::getSelectedTile() const
 	return std::nullopt;
 }
 
+bool CTileIconsSet::selectByTile(tile_index tileIdx)
+{
+	const auto prevSelected = selected;
+
+	auto tryToSelect = [this](tile_index idx) {
+		if (const auto it = ranges::find(icons, idx); it != icons.end()) {
+			select(std::distance(icons.begin(), it));
+			return true;
+		}
+		return false;
+	};
+	if (!tryToSelect(tileIdx)) {
+		tryToSelect(Map.Tileset.getFirstOfItsKindTile(tileIdx));
+	}
+	if (isSelected() && selected != prevSelected) {
+		const uint16_t k = selected / getDisplayedNum();
```

Comment:
Is there something which prevent `getDisplayedNum()` to be `0`?

Reply:
In theory it shouldn't be, but I agree - added handling of such a situation.